pandas/core/indexes/base.py

```diff
@@ -359,6 +360,10 @@ def _outer_indexer(
     _can_hold_na: bool = True
     _can_hold_strings: bool = True
 
+    # Whether this index is a NumericIndex, but not a Int64Index, Float64Index,
+    # UInt64Index or RangeIndex. Needed for backwards compat. Remove in pandas 2.0.
```

Comment:
can you create an issue for this removal (mark it for 2.0). can you also rename this attribute to something like
_is_backward_compat_public_numeric_index or similar (to inidicate what the purpose is)

Reply:
Ok done. See #42717.